code/controllers/subsystem/ghost_spawns.dm

```diff
@@ -0,0 +1,216 @@
+		var/alert_time = poll_time
+		var/alert_poll = P
+		if(current_alert && current_alert.timeout > (world.time + poll_time - world.tick_lag))
+			alert_time = current_alert.timeout - world.time + world.tick_lag
+			alert_poll = current_alert.poll
+
+		// Send them an on-screen alert
+		var/obj/screen/alert/notify_action/A = M.throw_alert(category, /obj/screen/alert/notify_action, timeout_override = alert_time, no_anim = TRUE)
+		if(!A)
+			continue
+
+		A.icon = ui_style2icon(M.client?.prefs.UI_style)
+		A.name = "Looking for candidates"
+		A.desc = "[question]\n\n(expires in [poll_time / 10] seconds)"
+		A.show_time_left = TRUE
+		A.poll = alert_poll
+
+		var/inherited_sign_up = FALSE
+		var/num_stack = 1
+		for(var/existing_poll in currently_polling)
+			var/datum/candidate_poll/P2 = existing_poll
+			if(P != P2 && P.hash == P2.hash)
+				// If there's already a poll for an identical mob type ongoing and the client is signed up for it, sign them up for this one
+				if(!inherited_sign_up && (M in P2.signed_up) && P.sign_up(M, TRUE))
+					A.display_signed_up()
+					inherited_sign_up = TRUE
+				// This number is used to display the number of polls the alert regroups
+				num_stack++
+		if(num_stack > 1)
+			A.display_stacks(num_stack)
+
+		// Image to display
+		var/image/I
+		if(source)
+			if(!ispath(source))
+				var/atom/S = source
+				var/old_layer = S.layer
+				var/old_plane = S.plane
+
+				S.layer = FLOAT_LAYER
+				S.plane = FLOAT_PLANE
+				A.overlays += S
+				S.layer = old_layer
+				S.plane = old_plane
+			else
+				I = image(source, layer = FLOAT_LAYER, dir = SOUTH)
+		else
+			// Just use a generic image
+			I = image('icons/effects/effects.dmi', icon_state = "static", layer = FLOAT_LAYER, dir = SOUTH)
+
+		if(I)
+			I.layer = FLOAT_LAYER
+			I.plane = FLOAT_PLANE
+			A.overlays += I
+
+		// Start processing it so it updates visually the timer
+		START_PROCESSING(SSprocessing, A)
+		A.process()
+
+	// Sleep until the time is up
+	UNTIL(P.finished)
+	return P.signed_up
+
+// Whether an observer is eligible to be an event mob
+/datum/controller/subsystem/ghost_spawns/proc/is_eligible(mob/M, role, antag_age_check, role_text, min_hours, check_antaghud)
+	. = FALSE
+	if(!M.key || !M.client)
+		return
+	if(role)
+		if(!(role in M.client.prefs.be_special))
+			return
+		if(antag_age_check)
+			if(!player_old_enough_antag(M.client, role))
+				return
+	if(role_text)
+		if(jobban_isbanned(M, role_text) || jobban_isbanned(M, "Syndicate"))
+			return
+	if(config.use_exp_restrictions && min_hours)
+		if(M.client.get_exp_type_num(EXP_TYPE_LIVING) < min_hours * 60)
+			return
+	if(check_antaghud && cannotPossess(M))
+		return
+
+	return TRUE
+
+// Called when polling is finished for a /datum/candidate_poll
+/datum/controller/subsystem/ghost_spawns/proc/polling_finished(datum/candidate_poll/P)
+	// Trim players who aren't eligible anymore
+	var/len_pre_trim = P.signed_up.len
```

Comment:
`length` instead of `.len`. Applies for other places as well. `length` is a bit faster and less runtime prone since it also catches null exceptions